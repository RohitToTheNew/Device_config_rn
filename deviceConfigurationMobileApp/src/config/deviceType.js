const deviceType = {
  ms700: 'MS-700',
  infoView: 'InfoviewPlayer',
  cza1300:'CZA1300'
};

export default deviceType;
