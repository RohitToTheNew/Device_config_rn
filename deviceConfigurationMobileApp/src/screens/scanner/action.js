const identifire = 'scanner-action log: ';
import { check, request, RESULTS, PERMISSIONS } from 'react-native-permissions';
import { RNCamera } from 'react-native-camera';
import utils from '../../utils';
import { translate } from '../../translations/translationHelper';
import RNFS from 'react-native-fs';

/**
 * On click Enter details manually and redirect user to Add device screen
 * @param {object} navigation it represents navigation object
 * @returns object
 */
function actionHandleManualDetails(navigation) {
  return () => {
    navigation.navigate('addDevice');
  };
}

/**
 * function to read local file data from storage
 * @returns file details read from storage
 */
const readFileFromLocalStorage = async () => {
  let result = RNFS.readFile(RNFS.DocumentDirectoryPath + '/test.txt', 'utf8');
  return result;
};

/**
 * function to check if the device details encoded into the QR code is having valid keys
 * @param {object} obj object containing the scanned data from QR code
 * @returns boolean indicating weather details are valid or not
 */
const isValidObject = (obj) =>
  obj?.serialNumber && obj?.serialNumber !== '' &&
  obj?.macAddress && obj?.macAddress !== '' &&
  obj?.deviceType && obj?.deviceType !== '';

/**
 * function to check if the device details encoded into the QR code is having valid keys
 * @param {object} obj object containing the scanned data from QR code
 * @returns boolean indicating weather details are valid or not
 */
const isValidDetails = (obj) =>
  obj?.serial && obj?.serial !== '' &&
  obj?.mac && obj?.mac !== '' &&
  obj?.device && obj?.device !== '';

/**
 * On successful scan
 * @param {object} navigation it represents navigation object
 * @param {function} setInvalidCode used to set invalid code boolean value
 * @param {object} e contains scan result
 */
function actionOnSuccess(navigation, setInvalidCode, e) {
  const typeToCheck = utils.isIOS ? 'org.iso.QRCode' : "QR_CODE"
  if (e.type === typeToCheck) {
    try {
      const scannedString = JSON.parse(e?.data);
      console.log('scanned string', scannedString)
      if (isValidObject(scannedString) || isValidDetails(scannedString)) {
        navigation.replace('addDevice', {
          mac: scannedString?.macAddress || scannedString?.mac,
          serialNumber: scannedString?.serialNumber || scannedString?.serial,
          deviceType: scannedString?.deviceType || scannedString?.device,
        });
      } else {
        setInvalidCode(true);
      }
    } catch (e) {
      setInvalidCode(true);
    }
  } else {
    setInvalidCode(true);
  }
}

/**
 * Click on close icon
 * @param {object} navigation it represents navigation object
 * @returns object
 */
function actionHandleClose(navigation) {
  return () => {
    navigation.goBack();
  };
}

/**
 * Click on flash icon
 * @param {boolean} isFlashOn used to identify if flash is on/off
 * @param {function} setFlash used to set flash on/off value
 * @param {function} setIsFlashOn used to set flash on value
 * @returns object
 */
function actionHandleFlash(isFlashOn, setFlash, setIsFlashOn) {
  return () => {
    if (isFlashOn) {
      setFlash(RNCamera.Constants.FlashMode.off);
      setIsFlashOn(false);
    } else {
      setFlash(RNCamera.Constants.FlashMode.torch);
      setIsFlashOn(true);
    }
  };
}

/**
 * To ask camera permission
 * @returns camera permission
 */
async function actionUseEffect() {

  try {
    const result = await check(PERMISSIONS.IOS.CAMERA);
    if (result !== RESULTS.GRANTED) {
      const askCameraPermission = await request(PERMISSIONS.IOS.CAMERA);
      return askCameraPermission;
    } else {
      utils.Log(utils.logType.log, identifire, 'Camera permisson granted');
    }
  } catch (Ex) {
    utils.showToast(translate('error'), translate('permissionsException'));
  }
}
export {
  actionHandleManualDetails,
  actionOnSuccess,
  actionHandleClose,
  actionUseEffect,
  actionHandleFlash,
};
