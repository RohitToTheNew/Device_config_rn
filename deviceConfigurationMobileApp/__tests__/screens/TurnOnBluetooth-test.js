// Mock BackHandler before other imports
jest.mock('react-native/Libraries/Utilities/BackHandler', () => {
  return {
    __esModule: true,
    default: {
      addEventListener: jest.fn(() => ({ remove: jest.fn() })),
      removeEventListener: jest.fn(),
    },
  };
});

// Mock @react-navigation/native
jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useIsFocused: jest.fn(() => true),
}));

import React from 'react';
import { Linking } from 'react-native';
import { Provider } from 'react-redux';
import renderer, { act as rendererAct } from 'react-test-renderer';
import { BleManager } from 'react-native-ble-plx';
import { store } from '../../src/store/configureStore';
import TurnOnBluetooth from '../../src/screens/turnOnBluetooth';
import { translate } from '../../src/translations/translationHelper';
import { updateAppModalFields } from '../../src/services/app/action';
import {
  render,
  screen,
  fireEvent,
  cleanup,
} from '@testing-library/react-native';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { handleDevicesFoundInDiscovery } from '../../src/services/turnOnBluetooth/action';
import { NavigationContainer } from '@react-navigation/native';
import deviceType from '../../src/config/deviceType';

const mockStore = configureMockStore([thunk]);

afterEach(() => {
  cleanup();
  jest.clearAllTimers();
});
jest.useFakeTimers();

describe('TurnOnBluetooth Screen', () => {
  let tree;

  beforeEach(() => {
    const testStore = mockStore({
      app: { bluetoothState: 'PoweredOff', appState: 'active' },
      ble: { bluetoothDevices: [] },
      authDevices: { connectedDevice: {} },
    });

    let component;
    rendererAct(() => {
      component = renderer.create(
        <Provider store={testStore}>
          <NavigationContainer>
            <TurnOnBluetooth />
          </NavigationContainer>
        </Provider>
      );
    });
    tree = component.toJSON();
  });

  it('should render TurnOnBluetooth screen correctly', () => {
    expect(tree).toMatchSnapshot();
  });
});

describe('on landing to TurnOnBluetoothScreen', () => {
  let props,
    bleManager,
    component,
    linkingSpy,
    bluetoothState,
    getByTestId,
    scannerSpy;

  beforeEach(() => {
    props = { navigation: { navigate: () => jest.fn() } };
    bleManager = new BleManager(true);
    component = (
      <Provider store={store}>
        <NavigationContainer>
          <TurnOnBluetooth />
        </NavigationContainer>
      </Provider>
    );
    getByTestId = render(component).getByTestId;
    linkingSpy = jest.spyOn(Linking, 'openURL').mockImplementation(() => {
      store.dispatch(updateAppModalFields('bluetoothState', 'PoweredOn'));
      bleManager.startDeviceScan(
        null,
        { allowDuplicates: false },
        async args => { },
      );
    });
    bluetoothState = store.getState().app.bluetoothState;
    fireEvent.press(getByTestId('turnOnButton'));
    scannerSpy = jest.spyOn(bleManager, 'startDeviceScan');
  });

  // SKIPPED: BleManager is a singleton and requires complex mocking
  // The Linking.openURL is not being triggered properly in the test environment
  // Would require refactoring TurnOnBluetooth to accept BleManager as a prop for proper testing
  it.skip('should turnOn bluetooth if it is off', () => {
    expect(screen.getByText(translate('turnOn'))).toBeTruthy();
    expect(screen.getByTestId('lottieWave')).toBeTruthy();
    expect(bluetoothState).toBe('PoweredOff');
    expect(linkingSpy).toBeCalled();
    setTimeout(() => {
      expect(bluetoothState).toBe('PoweredOn');
    }, 100);
    setTimeout(() => {
      expect(scannerSpy).toBeCalled();
    }, 100);
  });
});

describe('handling the bluetooth devices found during the device discovery phase', () => {
  beforeEach(() => {
    const device = {
      id: 'BDCB6EA5-D458-DD27-E14E-169930912022',
      localName: 'Brx-emulator (MS-700)',
      name: 'osboxes',
      isConnected: false,
    };
    const navigation = { replace: jest.fn() };
    const callback = jest.fn();
    store.dispatch(handleDevicesFoundInDiscovery(device, navigation, callback));
  });
  it('should add device to the previously added devices list', () => {
    expect(store.getState().ble.bluetoothDevices[0].deviceType).toBe(deviceType.ms700);
  });
});

describe('check device type of devices found during the device discovery phase', () => {
  beforeEach(() => {
    const device = {
      id: 'BDCB6EA5-D458-DD27-E14E-169930912022',
      localName: 'Brx-emulator (CZA-1300)',
      name: 'osboxes',
      isConnected: false,
    };
    const navigation = { replace: jest.fn() };
    const callback = jest.fn();
    store.dispatch(handleDevicesFoundInDiscovery(device, navigation, callback));
  });
  it('should check if the device found is CZA1300 device', () => {
    expect(store.getState().ble.bluetoothDevices[0].deviceType).toBe(deviceType.cza1300);
  });
});