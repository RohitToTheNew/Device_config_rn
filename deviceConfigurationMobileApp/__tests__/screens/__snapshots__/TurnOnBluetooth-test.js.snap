// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TurnOnBluetooth Screen should render TurnOnBluetooth screen correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <Image
    collapsable={false}
    resizeMode="stretch"
    source={
      {
        "default": "",
      }
    }
    style={
      {
        "height": 2000,
        "marginTop": -1000,
        "position": "absolute",
        "transform": [
          {
            "translateY": 0,
          },
        ],
        "width": 1502,
        "zIndex": -1,
      }
    }
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    hitSlop={
      {
        "bottom": 20,
        "left": 20,
        "right": 10,
        "top": 10,
      }
    }
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
        "position": "absolute",
        "right": 32,
        "top": 106,
      }
    }
  >
    <
      height={48}
      width={48}
    />
  </View>
  <View
    style={
      {
        "alignSelf": "center",
        "height": 600,
        "marginTop": 240,
        "width": 600,
      }
    }
    testID="lottieWave"
  >
    <View
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "height": 600,
          "opacity": 0,
          "width": 600,
          "zIndex": 55,
        }
      }
    >
      <<Lottie/>
        autoPlay={true}
        ref={
          {
            "current": null,
          }
        }
        resizeMode="cover"
        source={
          {
            "assets": [],
            "ddd": 0,
            "fr": 29.9700012207031,
            "h": 350,
            "ip": 0,
            "layers": [
              {
                "ao": 0,
                "bm": 0,
                "ddd": 0,
                "ind": 1,
                "ip": 0,
                "ks": {
                  "a": {
                    "a": 0,
                    "ix": 1,
                    "k": [
                      0,
                      0,
                      0,
                    ],
                  },
                  "o": {
                    "a": 0,
                    "ix": 11,
                    "k": 100,
                  },
                  "p": {
                    "a": 0,
                    "ix": 2,
                    "k": [
                      174.705,
                      174.649,
                      0,
                    ],
                  },
                  "r": {
                    "a": 0,
                    "ix": 10,
                    "k": 0,
                  },
                  "s": {
                    "a": 0,
                    "ix": 6,
                    "k": [
                      100,
                      100,
                      100,
                    ],
                  },
                },
                "nm": "icon",
                "op": 28.0000011404634,
                "shapes": [
                  {
                    "bm": 0,
                    "cix": 2,
                    "hd": false,
                    "it": [
                      {
                        "d": 1,
                        "hd": false,
                        "mn": "ADBE Vector Shape - Ellipse",
                        "nm": "Ellipse Path 1",
                        "p": {
                          "a": 0,
                          "ix": 3,
                          "k": [
                            0,
                            0,
                          ],
                        },
                        "s": {
                          "a": 0,
                          "ix": 2,
                          "k": [
                            67.909,
                            67.909,
                          ],
                        },
                        "ty": "el",
                      },
                      {
                        "bm": 0,
                        "c": {
                          "a": 0,
                          "ix": 4,
                          "k": [
                            1,
                            1,
                            1,
                            1,
                          ],
                        },
                        "hd": false,
                        "mn": "ADBE Vector Graphic - Fill",
                        "nm": "Fill 1",
                        "o": {
                          "a": 0,
                          "ix": 5,
                          "k": 100,
                        },
                        "r": 1,
                        "ty": "fl",
                      },
                      {
                        "a": {
                          "a": 0,
                          "ix": 1,
                          "k": [
                            0,
                            0,
                          ],
                        },
                        "nm": "Transform",
                        "o": {
                          "a": 0,
                          "ix": 7,
                          "k": 100,
                        },
                        "p": {
                          "a": 0,
                          "ix": 2,
                          "k": [
                            0.295,
                            0.351,
                          ],
                        },
                        "r": {
                          "a": 0,
                          "ix": 6,
                          "k": 0,
                        },
                        "s": {
                          "a": 0,
                          "ix": 3,
                          "k": [
                            143.334,
                            143.334,
                          ],
                        },
                        "sa": {
                          "a": 0,
                          "ix": 5,
                          "k": 0,
                        },
                        "sk": {
                          "a": 0,
                          "ix": 4,
                          "k": 0,
                        },
                        "ty": "tr",
                      },
                    ],
                    "ix": 1,
                    "mn": "ADBE Vector Group",
                    "nm": "Ellipse 1",
                    "np": 3,
                    "ty": "gr",
                  },
                ],
                "sr": 1,
                "st": 0,
                "ty": 4,
              },
              {
                "ao": 0,
                "bm": 0,
                "ddd": 0,
                "ind": 2,
                "ip": 0,
                "ks": {
                  "a": {
                    "a": 0,
                    "ix": 1,
                    "k": [
                      0,
                      0,
                      0,
                    ],
                  },
                  "o": {
                    "a": 0,
                    "ix": 11,
                    "k": 99,
                  },
                  "p": {
                    "a": 0,
                    "ix": 2,
                    "k": [
                      174.701,
                      174.644,
                      0,
                    ],
                  },
                  "r": {
                    "a": 0,
                    "ix": 10,
                    "k": 0,
                  },
                  "s": {
                    "a": 1,
                    "ix": 6,
                    "k": [
                      {
                        "i": {
                          "x": [
                            0.833,
                            0.833,
                            0.833,
                          ],
                          "y": [
                            0.833,
                            0.833,
                            0.833,
                          ],
                        },
                        "o": {
                          "x": [
                            0.167,
                            0.167,
                            0.167,
                          ],
                          "y": [
                            0.167,
                            0.167,
                            0.167,
                          ],
                        },
                        "s": [
                          101.276,
                          101.276,
                          100,
                        ],
                        "t": 22,
                      },
                      {
                        "s": [
                          118.381,
                          118.381,
                          100,
                        ],
                        "t": 27.0000010997325,
                      },
                    ],
                  },
                },
                "nm": "Supporting Circle",
                "op": 28.0000011404634,
                "shapes": [
                  {
                    "bm": 0,
                    "cix": 2,
                    "hd": false,
                    "it": [
                      {
                        "d": 1,
                        "hd": false,
                        "mn": "ADBE Vector Shape - Ellipse",
                        "nm": "Ellipse Path 1",
                        "p": {
                          "a": 0,
                          "ix": 3,
                          "k": [
                            0,
                            0,
                          ],
                        },
                        "s": {
                          "a": 0,
                          "ix": 2,
                          "k": [
                            67.909,
                            67.909,
                          ],
                        },
                        "ty": "el",
                      },
                      {
                        "bm": 0,
                        "c": {
                          "a": 0,
                          "ix": 4,
                          "k": [
                            0.517647087574,
                            0.631372570992,
                            0.752941191196,
                            1,
                          ],
                        },
                        "hd": false,
                        "mn": "ADBE Vector Graphic - Fill",
                        "nm": "Fill 1",
                        "o": {
                          "a": 0,
                          "ix": 5,
                          "k": 100,
                        },
                        "r": 1,
                        "ty": "fl",
                      },
                      {
                        "a": {
                          "a": 0,
                          "ix": 1,
                          "k": [
                            0,
                            0,
                          ],
                        },
                        "nm": "Transform",
                        "o": {
                          "a": 0,
                          "ix": 7,
                          "k": 100,
                        },
                        "p": {
                          "a": 0,
                          "ix": 2,
                          "k": [
                            0.295,
                            0.351,
                          ],
                        },
                        "r": {
                          "a": 0,
                          "ix": 6,
                          "k": 0,
                        },
                        "s": {
                          "a": 0,
                          "ix": 3,
                          "k": [
                            143.334,
                            143.334,
                          ],
                        },
                        "sa": {
                          "a": 0,
                          "ix": 5,
                          "k": 0,
                        },
                        "sk": {
                          "a": 0,
                          "ix": 4,
                          "k": 0,
                        },
                        "ty": "tr",
                      },
                    ],
                    "ix": 1,
                    "mn": "ADBE Vector Group",
                    "nm": "Ellipse 1",
                    "np": 3,
                    "ty": "gr",
                  },
                ],
                "sr": 1,
                "st": 0,
                "ty": 4,
              },
              {
                "ao": 0,
                "bm": 0,
                "ddd": 0,
                "ind": 3,
                "ip": 0,
                "ks": {
                  "a": {
                    "a": 0,
                    "ix": 1,
                    "k": [
                      0,
                      0,
                      0,
                    ],
                  },
                  "o": {
                    "a": 0,
                    "ix": 11,
                    "k": 100,
                  },
                  "p": {
                    "a": 0,
                    "ix": 2,
                    "k": [
                      174.651,
                      174.584,
                      0,
                    ],
                  },
                  "r": {
                    "a": 0,
                    "ix": 10,
                    "k": 0,
                  },
                  "s": {
                    "a": 1,
                    "ix": 6,
                    "k": [
                      {
                        "i": {
                          "x": [
                            0.833,
                            0.833,
                            0.833,
                          ],
                          "y": [
                            0.833,
                            0.833,
                            0.833,
                          ],
                        },
                        "o": {
                          "x": [
                            0.167,
                            0.167,
                            0.167,
                          ],
                          "y": [
                            0.167,
                            0.167,
                            0.167,
                          ],
                        },
                        "s": [
                          118.381,
                          118.381,
                          100,
                        ],
                        "t": 0,
                      },
                      {
                        "s": [
                          187.741,
                          187.741,
                          100,
                        ],
                        "t": 27.0000010997325,
                      },
                    ],
                  },
                },
                "nm": "Circle 1",
                "op": 28.0000011404634,
                "shapes": [
                  {
                    "bm": 0,
                    "cix": 2,
                    "hd": false,
                    "it": [
                      {
                        "d": 1,
                        "hd": false,
                        "mn": "ADBE Vector Shape - Ellipse",
                        "nm": "Ellipse Path 1",
                        "p": {
                          "a": 0,
                          "ix": 3,
                          "k": [
                            0,
                            0,
                          ],
                        },
                        "s": {
                          "a": 0,
                          "ix": 2,
                          "k": [
                            67.909,
                            67.909,
                          ],
                        },
                        "ty": "el",
                      },
                      {
                        "bm": 0,
                        "c": {
                          "a": 1,
                          "ix": 4,
                          "k": [
                            {
                              "i": {
                                "x": [
                                  0.833,
                                ],
                                "y": [
                                  0.833,
                                ],
                              },
                              "o": {
                                "x": [
                                  0.167,
                                ],
                                "y": [
                                  0.167,
                                ],
                              },
                              "s": [
                                0.517647087574,
                                0.631372570992,
                                0.752941250801,
                                1,
                              ],
                              "t": 0,
                            },
                            {
                              "s": [
                                0.29411765933,
                                0.46274510026,
                                0.643137276173,
                                1,
                              ],
                              "t": 27.0000010997325,
                            },
                          ],
                        },
                        "hd": false,
                        "mn": "ADBE Vector Graphic - Fill",
                        "nm": "Fill 1",
                        "o": {
                          "a": 0,
                          "ix": 5,
                          "k": 100,
                        },
                        "r": 1,
                        "ty": "fl",
                      },
                      {
                        "a": {
                          "a": 0,
                          "ix": 1,
                          "k": [
                            0,
                            0,
                          ],
                        },
                        "nm": "Transform",
                        "o": {
                          "a": 0,
                          "ix": 7,
                          "k": 100,
                        },
                        "p": {
                          "a": 0,
                          "ix": 2,
                          "k": [
                            0.295,
                            0.351,
                          ],
                        },
                        "r": {
                          "a": 0,
                          "ix": 6,
                          "k": 0,
                        },
                        "s": {
                          "a": 0,
                          "ix": 3,
                          "k": [
                            143.334,
                            143.334,
                          ],
                        },
                        "sa": {
                          "a": 0,
                          "ix": 5,
                          "k": 0,
                        },
                        "sk": {
                          "a": 0,
                          "ix": 4,
                          "k": 0,
                        },
                        "ty": "tr",
                      },
                    ],
                    "ix": 1,
                    "mn": "ADBE Vector Group",
                    "nm": "Ellipse 1",
                    "np": 3,
                    "ty": "gr",
                  },
                ],
                "sr": 1,
                "st": 0,
                "ty": 4,
              },
              {
                "ao": 0,
                "bm": 0,
                "ddd": 0,
                "ind": 4,
                "ip": 0,
                "ks": {
                  "a": {
                    "a": 0,
                    "ix": 1,
                    "k": [
                      0,
                      0,
                      0,
                    ],
                  },
                  "o": {
                    "a": 0,
                    "ix": 11,
                    "k": 100,
                  },
                  "p": {
                    "a": 0,
                    "ix": 2,
                    "k": [
                      174.446,
                      174.341,
                      0,
                    ],
                  },
                  "r": {
                    "a": 0,
                    "ix": 10,
                    "k": 0,
                  },
                  "s": {
                    "a": 1,
                    "ix": 6,
                    "k": [
                      {
                        "i": {
                          "x": [
                            0.833,
                            0.833,
                            0.833,
                          ],
                          "y": [
                            0.833,
                            0.833,
                            0.833,
                          ],
                        },
                        "o": {
                          "x": [
                            0.167,
                            0.167,
                            0.167,
                          ],
                          "y": [
                            0.167,
                            0.167,
                            0.167,
                          ],
                        },
                        "s": [
                          187.741,
                          187.741,
                          100,
                        ],
                        "t": 0,
                      },
                      {
                        "s": [
                          259.141,
                          259.141,
                          100,
                        ],
                        "t": 27.0000010997325,
                      },
                    ],
                  },
                },
                "nm": "Circle 2",
                "op": 28.0000011404634,
                "shapes": [
                  {
                    "bm": 0,
                    "cix": 2,
                    "hd": false,
                    "it": [
                      {
                        "d": 1,
                        "hd": false,
                        "mn": "ADBE Vector Shape - Ellipse",
                        "nm": "Ellipse Path 1",
                        "p": {
                          "a": 0,
                          "ix": 3,
                          "k": [
                            0,
                            0,
                          ],
                        },
                        "s": {
                          "a": 0,
                          "ix": 2,
                          "k": [
                            67.909,
                            67.909,
                          ],
                        },
                        "ty": "el",
                      },
                      {
                        "bm": 0,
                        "c": {
                          "a": 1,
                          "ix": 4,
                          "k": [
                            {
                              "i": {
                                "x": [
                                  0.833,
                                ],
                                "y": [
                                  0.833,
                                ],
                              },
                              "o": {
                                "x": [
                                  0.167,
                                ],
                                "y": [
                                  0.167,
                                ],
                              },
                              "s": [
                                0.29411765933,
                                0.462745130062,
                                0.643137276173,
                                1,
                              ],
                              "t": 0,
                            },
                            {
                              "s": [
                                0.105882354081,
                                0.321568638086,
                                0.54509806633,
                                1,
                              ],
                              "t": 27.0000010997325,
                            },
                          ],
                        },
                        "hd": false,
                        "mn": "ADBE Vector Graphic - Fill",
                        "nm": "Fill 1",
                        "o": {
                          "a": 0,
                          "ix": 5,
                          "k": 100,
                        },
                        "r": 1,
                        "ty": "fl",
                      },
                      {
                        "a": {
                          "a": 0,
                          "ix": 1,
                          "k": [
                            0,
                            0,
                          ],
                        },
                        "nm": "Transform",
                        "o": {
                          "a": 0,
                          "ix": 7,
                          "k": 100,
                        },
                        "p": {
                          "a": 0,
                          "ix": 2,
                          "k": [
                            0.295,
                            0.351,
                          ],
                        },
                        "r": {
                          "a": 0,
                          "ix": 6,
                          "k": 0,
                        },
                        "s": {
                          "a": 0,
                          "ix": 3,
                          "k": [
                            143.334,
                            143.334,
                          ],
                        },
                        "sa": {
                          "a": 0,
                          "ix": 5,
                          "k": 0,
                        },
                        "sk": {
                          "a": 0,
                          "ix": 4,
                          "k": 0,
                        },
                        "ty": "tr",
                      },
                    ],
                    "ix": 1,
                    "mn": "ADBE Vector Group",
                    "nm": "Ellipse 1",
                    "np": 3,
                    "ty": "gr",
                  },
                ],
                "sr": 1,
                "st": 0,
                "ty": 4,
              },
              {
                "ao": 0,
                "bm": 0,
                "ddd": 0,
                "ind": 5,
                "ip": 0,
                "ks": {
                  "a": {
                    "a": 0,
                    "ix": 1,
                    "k": [
                      0,
                      0,
                      0,
                    ],
                  },
                  "o": {
                    "a": 0,
                    "ix": 11,
                    "k": 100,
                  },
                  "p": {
                    "a": 0,
                    "ix": 2,
                    "k": [
                      174.446,
                      174.341,
                      0,
                    ],
                  },
                  "r": {
                    "a": 0,
                    "ix": 10,
                    "k": 0,
                  },
                  "s": {
                    "a": 1,
                    "ix": 6,
                    "k": [
                      {
                        "i": {
                          "x": [
                            0.833,
                            0.833,
                            0.833,
                          ],
                          "y": [
                            0.833,
                            0.833,
                            0.833,
                          ],
                        },
                        "o": {
                          "x": [
                            0.167,
                            0.167,
                            0.167,
                          ],
                          "y": [
                            0.167,
                            0.167,
                            0.167,
                          ],
                        },
                        "s": [
                          259.141,
                          259.141,
                          100,
                        ],
                        "t": 0,
                      },
                      {
                        "s": [
                          330.542,
                          330.542,
                          100,
                        ],
                        "t": 27.0000010997325,
                      },
                    ],
                  },
                },
                "nm": "Circle 3",
                "op": 28.0000011404634,
                "shapes": [
                  {
                    "bm": 0,
                    "cix": 2,
                    "hd": false,
                    "it": [
                      {
                        "d": 1,
                        "hd": false,
                        "mn": "ADBE Vector Shape - Ellipse",
                        "nm": "Ellipse Path 1",
                        "p": {
                          "a": 0,
                          "ix": 3,
                          "k": [
                            0,
                            0,
                          ],
                        },
                        "s": {
                          "a": 0,
                          "ix": 2,
                          "k": [
                            67.909,
                            67.909,
                          ],
                        },
                        "ty": "el",
                      },
                      {
                        "bm": 0,
                        "c": {
                          "a": 1,
                          "ix": 4,
                          "k": [
                            {
                              "i": {
                                "x": [
                                  0.833,
                                ],
                                "y": [
                                  0.833,
                                ],
                              },
                              "o": {
                                "x": [
                                  0.167,
                                ],
                                "y": [
                                  0.167,
                                ],
                              },
                              "s": [
                                0.105882361531,
                                0.321568638086,
                                0.54509806633,
                                1,
                              ],
                              "t": 0,
                            },
                            {
                              "s": [
                                0,
                                0.239215701818,
                                0.490196108818,
                                1,
                              ],
                              "t": 27.0000010997325,
                            },
                          ],
                        },
                        "hd": false,
                        "mn": "ADBE Vector Graphic - Fill",
                        "nm": "Fill 1",
                        "o": {
                          "a": 0,
                          "ix": 5,
                          "k": 100,
                        },
                        "r": 1,
                        "ty": "fl",
                      },
                      {
                        "a": {
                          "a": 0,
                          "ix": 1,
                          "k": [
                            0,
                            0,
                          ],
                        },
                        "nm": "Transform",
                        "o": {
                          "a": 0,
                          "ix": 7,
                          "k": 100,
                        },
                        "p": {
                          "a": 0,
                          "ix": 2,
                          "k": [
                            0.295,
                            0.351,
                          ],
                        },
                        "r": {
                          "a": 0,
                          "ix": 6,
                          "k": 0,
                        },
                        "s": {
                          "a": 0,
                          "ix": 3,
                          "k": [
                            143.334,
                            143.334,
                          ],
                        },
                        "sa": {
                          "a": 0,
                          "ix": 5,
                          "k": 0,
                        },
                        "sk": {
                          "a": 0,
                          "ix": 4,
                          "k": 0,
                        },
                        "ty": "tr",
                      },
                    ],
                    "ix": 1,
                    "mn": "ADBE Vector Group",
                    "nm": "Ellipse 1",
                    "np": 3,
                    "ty": "gr",
                  },
                ],
                "sr": 1,
                "st": 0,
                "ty": 4,
              },
            ],
            "markers": [],
            "meta": {
              "a": "",
              "d": "",
              "g": "LottieFiles AE 3.0.0",
              "k": "",
              "tc": "",
            },
            "nm": "Searching_Devices",
            "op": 28.0000011404634,
            "v": "4.8.0",
            "w": 350,
          }
        }
        style={
          {
            "height": "100%",
            "width": "100%",
          }
        }
        testID="wavelottie"
      />
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 80,
          "height": 160,
          "justifyContent": "center",
          "left": 222,
          "opacity": 1,
          "position": "absolute",
          "top": 222,
          "width": 160,
          "zIndex": 99,
        }
      }
    >
      < />
    </View>
  </View>
  <Text
    collapsable={false}
    style={
      {
        "alignSelf": "center",
        "color": "#FFFFFF",
        "fontFamily": "CenturyGothic",
        "fontSize": 34,
        "fontWeight": "700",
        "marginTop": 100,
        "opacity": 0,
      }
    }
  >
    [missing "en.searchingForNearby" translation]
  </Text>
  <View
    collapsable={false}
    style={
      {
        "marginTop": 86,
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        {
          "color": "#484949",
          "fontFamily": "CenturyGothic",
          "fontSize": 34,
          "fontWeight": "700",
          "textAlign": "center",
        }
      }
    >
      [missing "en.turnOnYour" translation]
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignSelf": "center",
          "marginTop": 82,
          "opacity": 1,
          "zIndex": 99,
        }
      }
      testID="turnOnButton"
    >
      <Text
        style={
          {
            "alignSelf": "center",
            "color": "#003D7D",
            "fontFamily": "CenturyGothic",
            "fontSize": 30,
            "fontWeight": "700",
          }
        }
      >
        [missing "en.turnOn" translation]
      </Text>
    </View>
  </View>
</View>
`;
